prusti-contracts/prusti-specs/src/common.rs

```diff
@@ -462,6 +462,18 @@ pub(crate) fn add_phantom_data_for_generic_params(item_struct: &mut syn::ItemStr
     }
 }
 
+/// Extracts the main/self type path from a type (e.g. `Foo` from `Foo`, `&Foo`, `[Foo]`, etc.)
```

Comment:
Hm, unsure I like this. This is a lossy conversion: is it used only to declare trait bounds or can it result in accidentally declaring the wrong extern spec?

Reply:
I agree that this has a bad smell, but I believe it's just used for rewriting `Self` and associated types to the original type/trait in `#[extern_spec]` and `#[refine_trait_spec]`, where the syntax is traversed until reaching a type path node and only that is replaced. Maybe I should add a note on what it's useful for?

Reply:
Actually, on second look, the suspicion was completely right. I think it would have been an issue and I just didn't think to test a spec on a non-type-path item involving a `Self` type anywhere 🤦
Addressed in the commit I just pushed; `extract_type_path` no longer exists!